apps/omg_watcher/lib/omg_watcher/db/repo.ex

```diff
@@ -36,6 +36,10 @@ defmodule OMG.Watcher.DB.Repo do
     chunk_size = @max_params_count |> div(entries |> hd |> fields_count)
 
     entries
+    |> Enum.map(fn entry ->
+      utc_now = OMG.Watcher.DateTime.utc_now()
+      Map.merge(entry, %{inserted_at: utc_now, updated_at: utc_now})
+    end)
```

Comment:
Is it worth adding a new `OMG.Watcher.DateTime` to do all the mocks if we could take one extra option instead? i.e.

```elixir
def insert_all_chunked(schema_or_source, entries, opts) do
  # ...
  {datetime, opts} = Keyword.pop(opts, :override_datetime, DateTime.utc_now())

  entries
  |> Enum.map(fn entry ->
    Map.merge(entry, %{inserted_at: datetime, updated_at: datetime})
  end)

  # ...
end
```

While arguably this changes the implementation to support a test, I think the mock introduces obscurity and complexity to this code by even larger & more hidden implementation in `OMG.Watcher.DateTime`.

Reply:
I don't understand how your code snippet this makes this code testable. Making a bare call to `DateTime.utc_now()` any where makes it impossible to control time. By wrapping it in a thin wrapper, we can either make the wrapper be a direct call-through to the native `DateTime` or we can direct it to a mock in our tests.

Also every time a test against this function runs it will potentially be a flaky test because the test_data will be random with every run (specifically DateTime.utc_now() will return a different value).

Can you explain a little bit more about your idea here?

Having said that I'm just about the mocking idea go and just write some flaky tests so this PR doesn't block anymore 😄 . If we're not ready to use mocks in our tests yet, then when can come back to this.